metrics/sink.go

```diff
@@ -86,10 +86,14 @@ func (t *TrendSink) Add(s Sample) {
 	t.Sum += s.Value
 	t.Avg = t.Sum / float64(t.Count)
 
+	if t.Count == 1 {
+		t.Max, t.Min = s.Value, s.Value
+		return
+	}
```

Comment:
nit: I feel this would be kind of nicer if it is `if t.Count ==0` at the very beginning with everything that happens on the *first* `Add` .

And then have the rest as is.

*Alternatively* we could have TrendSink be created with `NewTrendSink` which sets Max,Min to `-math.MaxFloat64` :thinking: this will make this logic not needed on each `Add`

Reply:
The idea of having a proper constructur `NewTrendSink` sounds good :+1:

Reply:
I also like this proposal, but I'm against doing the change in this PR (essentially in this release) for the following reasons:
1. We already have the Sink API under discussion at https://github.com/grafana/k6/pull/2898 and I would like to get a consensus there before. I need to restart that work.
2. I would like to have consistent Sink implementations so it would require adding the constructors also for the other sinks and it seems to me that it is too much for a last-minute change.

I add it as a suggestion to #2898 and will address it there.

Reply:
Make sense! Thanks for the explanation :+1: